codec-socks/src/main/java/io/netty/handler/codec/socks/SocksAuthRequestDecoder.java

```diff
@@ -28,39 +28,38 @@
  */
 public class SocksAuthRequestDecoder extends ReplayingDecoder<State> {
 
-    private SocksSubnegotiationVersion version;
-    private int fieldLength;
-    private String username;
-    private String password;
-    private SocksRequest msg = SocksCommonUtils.UNKNOWN_SOCKS_REQUEST;
-
     public SocksAuthRequestDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        String username = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
+                SocksSubnegotiationVersion version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
                 if (version != SocksSubnegotiationVersion.AUTH_PASSWORD) {
+                    out.add(SocksCommonUtils.UNKNOWN_SOCKS_REQUEST);
                     break;
                 }
                 checkpoint(State.READ_USERNAME);
             }
             case READ_USERNAME: {
-                fieldLength = byteBuf.readByte();
+                int fieldLength = byteBuf.readByte();
                 username = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
                 checkpoint(State.READ_PASSWORD);
             }
             case READ_PASSWORD: {
-                fieldLength = byteBuf.readByte();
-                password = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
-                msg = new SocksAuthRequest(username, password);
+                int fieldLength = byteBuf.readByte();
+                String password = SocksCommonUtils.readUsAscii(byteBuf, fieldLength);
+                out.add(new SocksAuthRequest(username, password));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Null passed for non-null parameter of new SocksAuthRequest(String, String) in io.netty.handler.codec.socks.SocksAuthRequestDecoder.decode(ChannelHandlerContext, ByteBuf, List) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)